pkg/auth/spire/delegate.go

```diff
@@ -247,7 +243,10 @@ func (s *SpireDelegateClient) initWatcher(ctx context.Context) (delegatedidentit
 
 	unixPath := fmt.Sprintf("unix://%s", s.cfg.SpireAdminSocketPath)
 
-	conn, err := grpc.Dial(unixPath, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	conn, err := grpc.Dial(unixPath, grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(20*1024*1024),
+			grpc.MaxCallSendMsgSize(20*1024*1024))) // setting this to 20MB to handle large bundles TODO: not make this the random chosen 20MB
```

Comment:
Typically we file an issue for followups like these to avoid having TODOs (at least new ones) in the code

Reply:
I forgot this one was in there, it was more of a note to myself when working on this :) 
Having a call with some SPIFFE maintainers later today where I hope to bring this up for advice on large number of workloads in general.

Reply:
Result was that changes are needed upstream, made an issue and linked it